jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -794,77 +794,69 @@ void java_object_factoryt::gen_nondet_struct_init(
   // * Not if the object has already been initialised by our caller, in which
   //   case they will set `skip_classid`
   // * Not if we're re-initializing an existing object (i.e. update_in_place)
-
-  bool skip_special_string_fields = false;
-
-  if(!is_sub &&
-     !skip_classid &&
-     update_in_place != update_in_placet::MUST_UPDATE_IN_PLACE)
+  // * Always if it has a string type (special case).
+
+  const bool is_char_sequence =
+    java_string_library_preprocesst ::implements_java_char_sequence(
```

Comment:
:pick: :question:  Is it customary to have a space before the double colon?

Reply:
No :p

Reply:
Huh, clang-format did this, not sure why! I'll fix it as CI is failing with a linting error anyway.